### PR TITLE
MDEV-31441 BLOB corruption on UPDATE of PRIMARY KEY with FOREIGN KEY

### PR DESCRIPTION
**Pull request by dr-m**
- [x] *The Jira issue number for this PR is: MDEV-31441*

## Description
`row_upd_clust_rec_by_insert()`: If we are resuming from a lock wait, reset the `disowned` flag of the BLOB pointers in `entry` that we copied from `rec` on which we had invoked `btr_cur_disown_inherited_fields()` before the lock wait started. In this way, the inserted record with the updated `PRIMARY KEY` value will have the BLOB ownership associated with itself, like it is supposed to be.

Note: If the lock wait had been aborted, then rollback would have invoked `btr_cur_unmark_extern_fields()` and no corruption would be possible.

## How can this PR be tested?
The test `innodb.foreign_key` was extended. It would be good to additionally test this with RQG with a broader selection of input. I do not think that it should make any difference if there is 1 or multiple BLOBs attached to the record.

The patch can be cherry-picked to 10.6 or later versions without conflicts. I think that this would be good to test in 10.6 or later as well, because the locking subsystem was refactored in 10.6.

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature and the PR is based against the latest MariaDB development branch.*
- [x] *This is a bug fix and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

## PR quality check
- [x] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/11.0/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [x] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

File: storage/innobase/row/row0upd.cc
Comment (vlad-lesin) [inline storage/innobase/row/row0upd.cc:2804]: The comment above says: "Skip delete marking clustered index and disowning its blobs." I would change it to "re-owning its blobs".

File: mysql-test/suite/innodb/t/foreign_key.test
Comment (vlad-lesin) [inline mysql-test/suite/innodb/t/foreign_key.test:974]: I would add a comment that the UPDATE is blocked on checking foreign keys reference constraints.

**Review by vlad-lesin — APPROVED** [review]
The fix looks good to me. I would add some comments, which would clarify the case, but the comments does not influence the code correctness, so it's up to @dr-m.

File: storage/innobase/row/row0upd.cc
Comment (dr-m) [reply storage/innobase/row/row0upd.cc:2804]: We will skip changing `rec`, but we must mark BLOBs in the `entry` as "owned", like they were on the previous call of `row_upd_clust_rec_by_insert()` that had returned `DB_LOCK_WAIT`. I will clarify the comment.

**Pull request merged**
